src/internal/pfsdb/repos.go

```diff
@@ -0,0 +1,363 @@
+			whereVal, limit, offset); err != nil {
+			return nil, errors.Wrap(err, "could not get repo page")
+		}
+		return page, nil
+	}
+	err := tx.SelectContext(ctx, &page, getRepoAndBranches+"GROUP BY repo.id, project.name ORDER BY repo.id ASC LIMIT $1 OFFSET $2 ;", limit, offset)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get repo page")
+	}
+	return page, nil
+}
+
+// CreateRepo creates an entry in the pfs.repos table.
+func CreateRepo(ctx context.Context, tx *pachsql.Tx, repo *pfs.RepoInfo) error {
+	if repo.Repo.Type == "" {
+		repo.Repo.Type = "unknown"
+	}
+	if repo.Repo.Project == nil {
+		repo.Repo.Project = &pfs.Project{Name: "default"}
+	}
+	_, err := tx.ExecContext(ctx, "INSERT INTO pfs.repos (name, type, project_id, description) "+
+		"VALUES ($1, $2::pfs.repo_type, (SELECT id from core.projects where name=$3), $4);",
+		repo.Repo.Name, repo.Repo.Type, repo.Repo.Project.Name, repo.Description)
+	if err != nil && IsErrRepoAlreadyExists(err) {
+		return ErrRepoAlreadyExists{Project: repo.Repo.Project.Name, Name: repo.Repo.Name}
+	}
+	return errors.Wrap(err, "create repo")
+}
+
+// DeleteRepo deletes an entry in the pfs.repos table.
+func DeleteRepo(ctx context.Context, tx *pachsql.Tx, repoProject, repoName, repoType string) error {
+	result, err := tx.ExecContext(ctx, "DELETE FROM pfs.repos "+
+		"WHERE project_id=(SELECT id FROM core.projects WHERE name=$1) AND name=$2 AND type=$3;", repoProject, repoName, repoType)
+	if err != nil {
+		return errors.Wrap(err, "delete repo")
+	}
+	rowsAffected, err := result.RowsAffected()
+	if err != nil {
+		return errors.Wrap(err, "could not get affected rows")
+	}
+	if rowsAffected == 0 {
+		return ErrRepoNotFound{Project: repoProject, Name: repoName, Type: repoType}
+	}
+	return nil
+}
+
+func GetRepoID(ctx context.Context, tx *pachsql.Tx, repoProject, repoName, repoType string) (pachsql.ID, error) {
+	row, err := getRepoRowByName(ctx, tx, repoProject, repoName, repoType)
+	if err != nil {
+		return pachsql.ID(0), err
+	}
+	return pachsql.ID(row.ID), nil
+}
+
+func getRepoRowByName(ctx context.Context, tx *pachsql.Tx, repoProject, repoName, repoType string) (*repoRow, error) {
+	row := &repoRow{}
+	if repoProject == "" {
+		repoProject = "default"
+	}
+	err := tx.QueryRowxContext(ctx, fmt.Sprintf("%s WHERE repo.project_id=(SELECT id from core.projects where name=$1) "+
+		"AND repo.name=$2 AND repo.type=$3 GROUP BY repo.id, project.name;", getRepoAndBranches), repoProject, repoName, repoType).StructScan(row)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, ErrRepoNotFound{Project: repoProject, Name: repoName, Type: repoType}
+		}
+		return nil, errors.Wrap(err, "scanning repo row")
+	}
+	return row, nil
+}
+
+// todo(fahad): rewrite branch related code during the branches migration.
+// GetRepo retrieves an entry from the pfs.repos table by using the row id.
+func GetRepo(ctx context.Context, tx *pachsql.Tx, id pachsql.ID) (*pfs.RepoInfo, error) {
```

Comment:
As mentioned previously having a distinct `RepoID` type would help ensure only the correct identifiers get passed to this function.